src/k8s-engine/wsl.ts

```diff
@@ -569,6 +570,26 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
     });
   }
 
+  /**
+   * Starts host-resolver's vsock-host process in wsl host
+   */
+  protected async startHostResolverHost() {
+    const exe = path.join(paths.resources, 'win32', 'host-resolver.exe');
+    const stream = await Logging['host-resolver'].fdStream;
+
+    return childProcess.spawn(exe, ['vsock-host'], {
+      stdio:       ['ignore', stream, stream],
+      windowsHide: true,
+    });
```

Comment:
Just doing a spawn isn't good enough — we need to restart the process if it dies unexpectedly. We have a `BackgroundProcess` class in this file that can manage that for you; please use it.